include/sparrow/layout/primitive_array.hpp

```diff
@@ -176,29 +180,56 @@ namespace sparrow
         const_value_iterator value_cbegin() const;
         const_value_iterator value_cend() const;
 
-        static arrow_proxy create_proxy(size_type n);
+        static arrow_proxy create_proxy(
+            size_type n,
+            std::optional<std::string_view>&& name = std::nullopt,
+            std::optional<std::string_view>&& metadata = std::nullopt
+        );
 
         template <validity_bitmap_input R = validity_bitmap>
-        static auto create_proxy(u8_buffer<T>&& data_buffer, R&& bitmaps = validity_bitmap{}) -> arrow_proxy;
+        static auto create_proxy(
+            u8_buffer<T>&& data_buffer,
+            R&& bitmaps = validity_bitmap{},
+            std::optional<std::string_view>&& name = std::nullopt,
+            std::optional<std::string_view>&& metadata = std::nullopt
```

Comment:
Same remark here for the 2 last arguments.

Reply:
done